api/core/models.py

```diff
@@ -83,8 +83,11 @@ class _BaseHistoricalModel(models.Model):
     class Meta:
         abstract = True
 
-    def get_change_details(self) -> typing.Optional[typing.List[ModelChange]]:  # type: ignore[return]
-        if self.history_type == "~":  # type: ignore[attr-defined]
+    def get_change_details(self) -> typing.List[ModelChange]:
```

Comment:
Just in case we could add a test to ensure it's covered
```
def test_get_change_details_handles_exception_prev_record_is_null():
    instance = HistoricalFeatureState(history_type="~")
    instance.prev_record = None 

    assert instance.get_change_details() == []
```
Ideally we could log when it happens but I let you decide if we will really track it or covering the case is enough